rust_icu_uloc/src/lib.rs

```diff
@@ -229,6 +233,58 @@ impl ULoc {
         )
     }
 
+    /// Implements `uloc_openKeywords()` from ICU4C.
+    pub fn keywords(&self) -> impl Iterator<Item = String> {
+        rust_icu_uenum::uloc_open_keywords(&self.repr)
+            .unwrap()
+            .map(|result| result.unwrap())
+    }
+
+    /// Implements `icu::Locale::getUnicodeKeywords()` from the C++ API.
+    pub fn unicode_keywords(&self) -> impl Iterator<Item = String> {
+        self.keywords().filter_map(|s| to_unicode_locale_key(&s))
+    }
+
+    /// Implements `uloc_getKeywordValue()` from ICU4C.
+    pub fn keyword_value(&self, keyword: &str) -> Result<Option<String>, common::Error> {
+        buffered_string_method_with_retry!(
+            buffered_string_keyword_value,
+            LOCALE_CAPACITY,
+            [
+                locale_id: *const raw::c_char,
+                keyword_name: *const raw::c_char,
+            ],
+            []
+        );
+        let locale_id = self.as_c_str();
+        let keyword_name =
+            ffi::CString::new(keyword).map_err(|_| common::Error::string_with_interior_nul())?;
+        buffered_string_keyword_value(
+            versioned_function!(uloc_getKeywordValue),
+            locale_id.as_ptr(),
+            keyword_name.as_ptr(),
+        )
+        .map(|value| if value.is_empty() { None } else { Some(value) })
+    }
+
+    /// Implements `icu::Locale::getUnicodeKeywordValue()` from ICU4C.
```

Comment:
This could probably also benefit from the above simplification.

Reply:
Same requirement to handle invalid keywords.